### ISSUE TITLE
`vscode`: fails to start on Mac OS

### ISSUE THREAD

**Issue by m0rg-dev** [NixOS/nixpkgs#234828]
### Describe the bug
Visual Studio Code 1.78.2 fails to start on Mac OS when installed through Nix:

<img width="372" alt="image" src="https://github.com/NixOS/nixpkgs/assets/38578268/c91c36c7-689e-4d5f-9522-9236c7abbf53">

`spctl` doesn't like it:

```
$ spctl --verbose --assess --type execute /nix/store/0pwc8488074zchjf00k28dxknszifsla-vscode-1.78.2/Applications/Visual\ Studio\ Code.app
/nix/store/0pwc8488074zchjf00k28dxknszifsla-vscode-1.78.2/Applications/Visual Studio Code.app: a sealed resource is missing or invalid
```

Tried two different machines, one aarch64 (Studio, Ventura 13.3.1a) and one x86_64 (2019 MBP, Ventura 13.3) with the same result. Downloading VSCode 1.78.2 directly from Microsoft and running that works fine so I'm inclined to assume it's some sort of issue with the Nix package.

### Steps To Reproduce
```
$ nix-shell -p vscode
[nix-shell:~]$ code
Killed: 9
```

### Notify maintainers

@eadwu, @maxeaubrey, @bobby285271

### Metadata
From the Mac Studio:

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.4.0, macOS 13.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixpkgs"`
 - channels(morgan): `"home-manager, nixpkgs"`
 - nixpkgs: `/Users/morgan/.nix-defexpr/channels/nixpkgs`
```

**Comment by m0rg-dev** [comment]
also gonna investigate this one myself but posting an issue in case someone already knows what's going on here

**Comment by m0rg-dev** [comment]
Looks like #151918 (@Enzime) is probably the culprit here. Just built `vscode` from a local checkout of the commit before that and it works fine (and that also packages VSCode 1.78.2).

That PR looks like it's updating the `vscode` package to replace a bundled utility in the application, which is definitely going to interfere with macOS code signing (presumably the system `rg` binary is the "missing or invalid" sealed resource here). Unfortunately I'm not up to speed on macOS code signing practices to the point of knowing what to do about it...

**Comment by Enzime** [comment]
I'll have a look into code signing when I have a chance, for reference, if macOS ever complains about unsigned binaries like this:

You can go into System Preferences -> Privacy & Security and if you scroll down there should be a button to click `Open Anyway` which will make macOS remember that binary is okay

https://support.apple.com/en-au/guide/mac-help/mh40616/mac

**Comment by m0rg-dev** [comment]
I had a chat with some people on Matrix about this and it's unlikely we'll be able to deal with it from the package side - it's actually a "notarization" issue rather than a "signing" issue

**Comment by wegank** [comment]
Should be closed by #234871.
